CSS dataset format (import only) (#685)
openvinotoolkit/datumaro: site/content/en/docs/formats/common_semantic_segmentation.md

---
title: 'Common Semantic Segmentation'
linkTitle: 'Common Semantic Segmentation'
description: ''
weight: 2
---

## Format specification

CSS format specification is available [here](https://github.com/openvinotoolkit/open_model_zoo/blob/master/tools/accuracy_checker/openvino/tools/accuracy_checker/annotation_converters/README.md#supported-converters).

Supported annotation types:
- `Masks`

## Import Common Semantic Segmentation dataset

A Datumaro project with a CSS source can be created in the following way:

``` bash
datum create
datum import --format common_semantic_segmentation <path/to/dataset>
```

> Extra options for adding a source in the COCO format:

Reviewer: ```suggestion
Extra import options:
```

COCO (panoptic) doesn't seem to be related to this format, because it stores masks in a different encoding. Also, they won't be located by this format, if the dataset follows the default Datumaro COCO layout.

- `--image-prefix IMAGE_PREFIX` allow to import dataset with custom image prefix
  (by default '')
- `--mask-prefix MASK_PREFIX` allow to import dataset with custom mask prefix
  (by default '')

CSS dataset directory should have the following structure:

<!--lint disable fenced-code-flag-->
```
└─ Dataset/
    ├── dataset_meta.json # a list of labels
    ├── images/
    │   ├── <img1>.png
    │   ├── <img2>.png
    │   └── ...
    └── masks/
        ├── <img1>.png
        ├── <img2>.png
        └── ...
```

> To add classes, you should use [`dataset_meta.json`](/docs/user-manual/supported_formats/#dataset-meta-file).

Reviewer: ```suggestion
To describe classes and colors, you should use [`dataset_meta.json`](/docs/user-manual/supported_formats/#dataset-meta-file).
```

To make sure that the selected dataset has been added to the project, you can
run `datum project info`, which will display the project information.

## Export to other formats

Datumaro can convert a CSS dataset into any other format [Datumaro supports](/docs/user-manual/supported_formats/).
To get the expected result, convert the dataset to formats

> that support the segmentation task (e.g. PASCAL VOC, CamVid, CItyscapes, etc.)

Reviewer: ```suggestion
that support the segmentation task (e.g. PASCAL VOC, CamVid, Cityscapes, etc.)
```
